### PR TITLE
[storage] lazy commit step 1: separate commits to two dbs

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

according to lazy commit, we are going to separate commits to ledger db and state merkle db separately with different phases. This PR is a preparation for that in terms of API separation.

### CODE REVIEW

File: storage/aptosdb/src/lib.rs [hunk 1]
```diff
@@ -718,8 +694,10 @@ impl AptosDB {
             self.ledger_store
                 .put_transaction_infos(first_version, &txn_infos, cs)?
         };
+        self.state_store
+            .put_value_sets(state_updates_vec, first_version, cs)?;
```
Comment (msmouse) [inline storage/aptosdb/src/lib.rs:698]: Measure this. Why moved to last?

File: execution/executor-types/src/lib.rs [hunk 2]
```diff
@@ -83,6 +83,12 @@ pub trait ChunkExecutorTrait: Send + Sync {
     fn reset(&self) -> Result<()>;
 }
 
+pub struct SnapshotDelta {
```
Comment (sitalkedia) [inline execution/executor-types/src/lib.rs:86]: nit: `SnapshotDelta` is a very generic term and doesn't signfy what delta are we talking about. May be `StateSnapshotDelta` or `StateDelta` is a better name?

File: storage/storage-interface/src/lib.rs [hunk 3]
```diff
@@ -664,11 +664,11 @@ pub trait DbWriter: Send + Sync {
...
 
-    /// Persists merklized states as authenticated state snapshot.
-    /// See [`AptosDB::save_state_snapshot`].
+    /// Persists merklized states as authenticated state checkpoint.
+    /// See [`AptosDB::save_state_checkpoint`].
     ///
-    /// [`AptosDB::save_state_snapshot`]: ../aptosdb/struct.AptosDB.html#method.save_state_snapshot
-    fn save_state_snapshot(
+    /// [`AptosDB::save_state_checkpoint`]: ../aptosdb/struct.AptosDB.html#method.save_state_checkpoint
+    fn save_state_checkpoint(
```
Comment (msmouse) [inline storage/storage-interface/src/lib.rs:671]: I'm actually okay with calling it a snapshot. Let's keep it consistent with your `xDelta` struct at least.

**Review by msmouse — APPROVED** [review]

File: execution/executor-types/src/lib.rs (hunk 2 above)
Comment (lightmark) [reply execution/executor-types/src/lib.rs:86]: changed to state_snapshot_delta

**Pull request merged**